src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs

```diff
@@ -83,6 +92,11 @@ internal BlameCollector(IBlameReaderWriter blameReaderWriter, IProcessDumpUtilit
             DataCollectionLogger logger,
             DataCollectionEnvironmentContext environmentContext)
         {
+            if (string.Equals(Environment.GetEnvironmentVariable("DEBUG_BLAMEDATACOLLECTOR"), "true", StringComparison.OrdinalIgnoreCase))
```

Comment:
If we are providing debug for blame directly, then we also need to mention what process the user has to attach to.
Also how does this work on machines with non VS?

Can we follow [https://github.com/microsoft/vstest/blob/1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/vstest.console/Program.cs#L22](https://github.com/microsoft/vstest/blob/1e37ee32dbba48ad797b03bff0058c5c60a3a770/src/vstest.console/Program.cs#L22)

Reply:
This directly launches the default windows debug window in case of no VS i believe. And then you can choose what program to use to debug.

Reply:
The problem is this guy does not yet have a handle to write to the main console. so printing and waiting for a debugger doesn't make sense afaik.

Reply:
This seems redundant, please remove.
We can change the debug logic at the datacollector process in that case.

---
In reply to: [310884661](https://github.com/microsoft/vstest/pull/2110#discussion_r310884661) [](ancestors = 310884661)